### ISSUE TITLE
[BUG] fvm not working on windows

### ISSUE THREAD

**Issue by Azkarell** [nvim-flutter/flutter-tools.nvim#343]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

Running :FlutterRun with fvm enabled throws ENOENT: no such file or directory in windows
![grafik](https://github.com/akinsho/flutter-tools.nvim/assets/12845778/078014a4-66b0-4f33-b10d-f0bf9007f68f)
although path is detected correctly

### Expected Behavior

Running :FlutterRun in windows should run flutter

### Steps To Reproduce

my conifg with lazy:
{
    'akinsho/flutter-tools.nvim',
    lazy = false,
    dependencies = { 'nvim-lua/plenary.nvim' },
    config = function()
      require('flutter-tools').setup {
        fvm = true,
        -- debugger = {
        --   enabled = true,
        --   run_via_dap = true,
        --   register_configurations = function(paths)
        --     local file = io.open('.vscode/launch.json', 'r')
        --     if file ~= nil then
        --       io.close(file)
        --       require('dap.ext.vscode').load_launchjs()
        --     end
        --   end,
        -- },
      }
      require('telescope').load_extension 'flutter'
    end,
  },

### Environment

```Markdown
- OS: Windows 11 Pro
- Flutter version: fvm mit flutter 3.13.9
- Is flutter in $PATH: yes
- neovim version: 0.9.5
```

### Anything else?

I tried to find out what is happening and it might not even be the fault of this tool but i am quite new to neovim and just evaluating if i want to continue using it. what i found that just the line
:lua =vim.uv.spawn("C:\\**\\**\\fvm\\versions\\3.13.9\\bin\\flutter", { { "run" }}) already fails

just running "C:\\**\\**\\fvm\\versions\\3.13.9\\bin\\flutter" in pwsh or cmd or even with Start-Process is working.

**Comment by sidlatau** [comment]
Does it work if you run `fvm` outside neovim: `fvm flutter run`?

**Comment by Azkarell** [comment]
yes everything works as expected outside of neovim. only this plugin/neovim can't start it.

**Comment by sidlatau** [comment]
Please check the latest plugin version,https://github.com/akinsho/flutter-tools.nvim/commit/d8f2eac1734e0e68050bc57600e5f2ba775b1ec4 commit may help. If there is still a problem, please reopen the issue.

**Comment by AyushUlric** [comment]
Facing this issue on windows with latest commits. This happens when I set fvm = true. However I've noticed that if i set fvm = false and add flutterpath = 'C:\\Users\\daban\\fvm\\versions\\3.19.5\\bin\\flutter.bat' everthing works fine. Any thoughts?
![image](https://github.com/user-attachments/assets/09aa8686-ef63-4d4a-a331-fe9a3078f089)

**Comment by sidlatau** [comment]
Yea, it looks like missing `.bat` suffix is a problem. I will reopen this case to fix it.

**Comment by sidlatau** [comment]
I do not have Windows, so I am not sure if the path is the same as in macOS. @AyushUlric, if you have time please check if this works for you https://github.com/akinsho/flutter-tools.nvim/pull/386? The setup would be: 
```lua
return {
  "akinsho/flutter-tools.nvim",
  branch = "fvm-on-windows",
...
}
```

**Comment by AyushUlric** [comment]
@sidlatau, didn't work, same error. I hope I didn't mess up the setup as I am fairly new to nvim so attaching my setup code below.

![image](https://github.com/user-attachments/assets/aa2e281b-8bf7-41f7-a3a0-773fb82ea45e)

![image](https://github.com/user-attachments/assets/3a643fce-f029-4f13-998a-254e1f9090f5)

I then updated all my plugins from the Lazy menu and restarted nvim.

I think on windows flutter and flutter.bat will always be in the same directory and the current path correctly points to the flutter file, hope that helps.

**Comment by sidlatau** [comment]
@AyushUlric thanks for testing. I see you are using global path to `fvm` flutter. But there should be local one in your project, in `.fvm` directory:
<img width="485" alt="image" src="https://github.com/user-attachments/assets/013ef928-7bbc-4c5e-ab9b-d01e15b25007">
Plugin constructs path `.fvm/flutter_sdk/bin/flutter.bat`. Does this file exists in your project? Did you run `fvm use <flutter version>` in your project directory?

**Comment by AyushUlric** [comment]
@sidlatau Yes I did run the fvm use command and I have the .fvm directory in my project root. But I noticed that flutter_sdk directory is actually a shotcut which points to the global fvm path. I checked all my fvm projects and it seems to be common in all of them.

![image](https://github.com/user-attachments/assets/d5b098b7-3992-4015-8023-ab1df19c1fb5)
![image](https://github.com/user-attachments/assets/77fd4ff2-cafb-415d-a190-d588ec20e617)

Maybe it's how it is on Windows? 

BTW I added the global path because thats what showed up in error logs with path not found.

![image](https://github.com/user-attachments/assets/b377c1d6-b91a-4e93-9da7-308ef84b512c)

**Comment by sidlatau** [comment]
Hopefully should be fixed with https://github.com/nvim-flutter/flutter-tools.nvim/pull/450
